File: src/main/java/gabrielruiu/packt/email/NoopEmailSender.java
package gabrielruiu.packt.email;

import gabrielruiu.packt.BookSummary;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

/**
 * @author Gabriel Mihai Ruiu (gabriel.ruiu@mail.com)
 */
@Profile("noop")
@Component
public class NoopEmailSender implements EmailSender {

    @Override
    public void sendEmailWithBookSummary(BookSummary bookSummary) {

    }
}

File: src/main/kotlin/gabrielruiu/packt/email/NoopEmailSender.kt
package gabrielruiu.packt.email

import gabrielruiu.packt.BookSummary
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Profile("noop")
@Component
open class NoopEmailSender : EmailSender{

    override fun sendEmailWithBookSummary(bookSummary: BookSummary) {
    }
}